app/src/main/java/com/mapbox/maps/testapp/utils/SimulateRouteLocationProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.maps.testapp.utils
+
+import android.os.Handler
+import android.os.Looper
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.maps.plugin.locationcomponent.LocationConsumer
+import com.mapbox.maps.plugin.locationcomponent.LocationProvider
+import com.mapbox.turf.TurfMeasurement
+import java.util.concurrent.*
+
+/**
+ * A location provider implementation that takes in a line string as route and animate the location
+ * updates along the route.
+ */
+class SimulateRouteLocationProvider(
+  val route: LineString,
+) : LocationProvider {
+  private val locationConsumers = CopyOnWriteArraySet<LocationConsumer>()
+  private val handler = Handler(Looper.getMainLooper())
+  override fun registerLocationConsumer(locationConsumer: LocationConsumer) {
+    locationConsumers.add(locationConsumer)
+    emitFakeLocations()
+  }
+
+  override fun unRegisterLocationConsumer(locationConsumer: LocationConsumer) {
+    locationConsumers.remove(locationConsumer)
+  }
+
+  private val pointList = route.coordinates().toMutableList()
+  private val iterator = pointList.iterator()
+  var lastLocation: Point = iterator.next()
```

Comment:
is it okay if `.next()` can throw exception if there is not elements?

Reply:
Added check before using next()